scipy/stats/kde.py

```diff
@@ -241,28 +242,11 @@ def evaluate(self, points):
                     self.d)
                 raise ValueError(msg)
 
-        result = zeros((m,), dtype=float)
-
-        whitening = linalg.cholesky(self.inv_cov)
-        scaled_dataset = dot(whitening, self.dataset)
-        scaled_points = dot(whitening, points)
-
-        if m >= self.n:
-            # there are more points than data, so loop over data
-            for i in range(self.n):
-                diff = scaled_dataset[:, i, newaxis] - scaled_points
-                energy = sum(diff * diff, axis=0) / 2.0
-                result += self.weights[i]*exp(-energy)
-        else:
-            # loop over points
-            for i in range(m):
-                diff = scaled_dataset - scaled_points[:, i, newaxis]
-                energy = sum(diff * diff, axis=0) / 2.0
-                result[i] = sum(exp(-energy)*self.weights, axis=0)
-
-        result = result / self._norm_factor
-
-        return result
+        result = _gaussian_kernel_estimate(
+            self.dataset.T.astype(float), self.weights[:, None].astype(float),
+            points.T.astype(float), self.inv_cov.astype(float)
```

Comment:
any concerns about increased space complexity with i.e., extra copies here?

Reply:
Good point. Could use fused types in cython to get around that issue?